src/factories/ResourceFactory.js

```diff
@@ -0,0 +1,111 @@
+import { itemType } from '../enums'
+import {
+  RESOURCE_SPAWN_CHANCE,
+  ORE_SPAWN_CHANCE,
+  COAL_SPAWN_CHANCE,
+  STONE_SPAWN_CHANCE,
+} from '../constants'
+import { randomChoice } from '../common/utils'
+
+import OreFactory from './OreFactory'
+import CoalFactory from './CoalFactory'
+import StoneFactory from './StoneFactory'
+
+/**
+ * Used for spawning mined resources
+ * @constructor
+ */
+export default class ResourceFactory {
+  constructor() {
+    this.resourceOptions = [
+      { weight: ORE_SPAWN_CHANCE, name: itemType.ORE },
+      { weight: COAL_SPAWN_CHANCE, name: itemType.FUEL },
+      { weight: STONE_SPAWN_CHANCE, name: itemType.STONE },
```

Comment:
Would it make sense for the `name`s to be called `itemType` here?

Reply:
good catch. when i started this i was thinking `randomChoices' would always expect "weight" and "name" but didn't end up sticking to that. updating